extensions/redis-client/deployment/src/main/java/io/quarkus/redis/client/deployment/DevServicesProcessor.java

```diff
@@ -0,0 +1,204 @@
+package io.quarkus.redis.client.deployment;
+
+import static io.quarkus.redis.client.runtime.RedisClientUtil.isDefault;
+
+import java.io.Closeable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.OptionalInt;
+
+import org.jboss.logging.Logger;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.utility.DockerImageName;
+
+import io.quarkus.bootstrap.classloading.QuarkusClassLoader;
+import io.quarkus.deployment.IsDockerWorking.IsDockerRunningSilent;
+import io.quarkus.deployment.IsNormal;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.RunTimeConfigurationDefaultBuildItem;
+import io.quarkus.deployment.builditem.ServiceStartBuildItem;
+import io.quarkus.redis.client.deployment.RedisBuildTimeConfig.DevServiceConfiguration;
+import io.quarkus.redis.client.deployment.devservices.DevServicesRedisResultBuildItem;
+import io.quarkus.redis.client.deployment.devservices.DevServicesRedisResultBuildItem.Result;
+import io.quarkus.redis.client.runtime.RedisClientUtil;
+import io.quarkus.redis.client.runtime.RedisConfig;
+import io.quarkus.runtime.LaunchMode;
+import io.quarkus.runtime.configuration.ConfigUtils;
+
+public class DevServicesProcessor {
+    private static final Logger log = Logger.getLogger(DevServicesProcessor.class);
+    private static final String REDIS_6_ALPINE = "redis:6-alpine";
+    private static final int REDIS_EXPOSED_PORT = 6379;
+    private static final String REDIS_SCHEME = "redis://";
+    private static final String QUARKUS = "quarkus.";
+    private static final String DOT = ".";
+    private static volatile List<Closeable> closeables;
+    private static volatile Map<String, DevServiceConfiguration> capturedDevServicesConfiguration;
+    private static volatile boolean first = true;
+
+    @BuildStep(onlyIfNot = IsNormal.class, onlyIf = IsDockerRunningSilent.class)
+    public DevServicesRedisResultBuildItem startRedisContainers(LaunchModeBuildItem launchMode,
```

Comment:
At some point in the future, it would probably make sense to extract as much common code as we can from this and mongo

Reply:
+1 I agree. Especially with other extensions leveraging the same mechanism https://github.com/quarkusio/quarkus/pull/17083